### PR TITLE
feat: notify user when they earn a new badge. Closes #339

### LINKED ISSUE #339: As a contributor I want to know when I've earned a badge
Emails will help create excitement.

### PR DESCRIPTION
**Pull request by madzimai**
Hey team. First PR here so be nice :see_no_evil: (j/k, fire away).

I added placeholder content for the email. You'll have to tell me the exact content you want in here.

### CODE REVIEW

**Comment by goose-life** [comment]
@ro6ley you've already worked on other notification emails right? 
if the logic looks good to you I'll double check the wording of the email before merging

**Comment by ro6ley** [comment]
@goose-life yes, I have worked on other notification emails, let me review.

👋🏽 @madzimai

File: indigo_social/badges.py [hunk 1]
```diff
@@ -58,6 +67,25 @@ def revoke(self, user):
         """
         pass
 
+    def send_templated_email(self, template_name, recipient, context, **kwargs):
```
Comment (ro6ley) [inline indigo_social/badges.py]: This is already implemented in the Notifier class, (see: https://github.com/laws-africa/indigo/blob/ft-annotation-reply-email-notification/indigo_app/notifications.py), no need to redo it here again. 

Pinax Badges send a `badge_awarded` signal that you can receive in the Notifier class and dispatch the email from there, like we do for other event-based notifications. Also, these badge notifications will be used by another part of this project.

You can refer to this, as I have done email notifications for new replies to an annotation: https://github.com/laws-africa/indigo/blob/5f33c59c39b951a9464001edcdb7e9093da26fb8/indigo_app/notifications.py#L143

File: indigo_social/badges.py [hunk 2]
```diff
@@ -39,6 +42,12 @@ def award(self, user, **state):
         """
         if self.can_award(user, **state):
             self.grant(user)
+            self.send_templated_email('badge_awarded', user, {
```
Comment (ro6ley) [inline indigo_social/badges.py]: This will not be needed.

**Comment by longhotsummer** [comment]
This has been done.

**Pull request closed without merging**